### PR TITLE
delete redundant getUser method and redundant '/' in routes

### PR DESCRIPTION
**Pull request by usborn116**
### Removed getUser
The getUser method had the same logic as getData, so I removed that function and replaced all instances of it to use getData instead.

While I was editing useEffects to use the getData function, I also removed redundant '/' marks where not needed.

### CODE REVIEW

File: app/javascript/components/Games.jsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const Games = ({endpoint, loading, user, homeError = null}) => {
     const [numCategories, setNumCategories] = useState(0)
 
     useEffect(() => {
-        getData(`/${user ? 'user_games' : 'games'}${search ? `?name=${search}`: ''}`, setData, homeError || setError)
+        getData(`${user ? 'user_games' : 'games'}${search ? `?name=${search}`: ''}`, setData, homeError || setError)
```
Comment (nicholaskillin) [inline app/javascript/components/Games.jsx:26]: I see that you are simply removing a `/` from this endpoint string, so perhaps this is outside of the scope of this refactor. However, I wanted to mention that this string being built takes some time to wrap my mind around. It might be best to pull some of these things out into namable variables to help make parsing this string easier on the reader.
```suggestion
        const url = user ? 'user_games' : 'games'
        if (search) {
          url += `?name=${search}`
        }
        
        getData(url, setData, homeError || setError)
```

**Review by nicholaskillin — COMMENTED** [review]
Overall this looks good. I'm all for standardizing on how you want to fetch data on the front end of things.

I noticed right at the end of looking at this that you seem to have a `useSetUser` hook that can fetch the user and return it to any components needing it. Out of curiosity, is there a reason these other components aren't using that hook?

File: app/javascript/components/Games.jsx (hunk 1 above)
Comment (usborn116) [reply app/javascript/components/Games.jsx:26]: that's helpful input, thanks! I definitely see how this can be tedious to understand as written; I'll look into clarifying this in a future PR
